Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -187,13 +193,12 @@ public List<TheoreticalFragmentIon> GetTheoreticalFragments(List<ProductType> pr
                 }
 
                 List<ProductType> temp = new List<ProductType> { productType };
-                TerminusType terminusType = ProductTypeMethods.IdentifyTerminusType(temp);
-
-                var productMasses = new CompactPeptide(this, terminusType).ProductMassesMightHaveDuplicatesAndNaNs(temp);
+                
+                var productMasses = new CompactPeptide(this, fragmentationTerminus, dissociationType).ProductMassesMightHaveDuplicatesAndNaNs(temp);
 
                 for (int i = 0; i < productMasses.Length; i++)
                 {
-                    theoreticalFragmentIons.Add(new TheoreticalFragmentIon(productMasses[i], double.NaN, 1, productType, i + ionNumberAdd));
+                    theoreticalFragmentIons.Add(  new TheoreticalFragmentIon(productMasses[i], double.NaN, 1, productType, i + ionNumberAdd));
```

Comment:
need to make an issue for this; if neutral losses are part of this modified peptide, the fragment numbering will be wrong (e.g., should be y1 and y1, not y1 and y2)